specs/plasma.md

```diff
@@ -0,0 +1,199 @@
+# Plasma mode
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [DA Storage](#da-storage)
+- [Input Commitment](#input-commitment)
+- [Data Availability Challenge Contract](#data-availability-challenge-contract)
+  - [Parameters](#parameters)
+- [Derivation](#derivation)
+- [Safety and Finality](#safety-and-finality)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Using [Plasma][vitalikblog] mode means switching to an offchain data availability provider.
+The protocol guarantees availability with a challenge contract on L1 where users
+can challenge the availability of input data used to derive the chain.
+Challenging an input commitment means forcing providers to submit the input data on chain
+to prove the data is available during a time period. We call the period during which any party
+can challenge a commitment `challenge_window` and the period during which any party
+can submit the input data to resolve the challenge `resolve_window`.
+If a challenge is not resolved by the time `resolve_window` of L1 blocks have passed,
+chain derivation must be reset, omitting the input data when rederiving therefore causing a reorg.
+
+## DA Storage
+
+Input data is uploaded to the storage layer via plain HTTP calls to the DA storage service.
+This service is horizontally scalable and concerned with replicating the data across prefered storage layers
+such as IPFS or any S3 compatible storage. Input data is content addressed by its hash in the request url
+so responses are easily cachable.
```

Comment:
Is this a required property of the spec? This describes desirable properties of a DA Storage, but presumably any reasonably available technology would work as well, right? I'm trying to figure out if there's a reason something like FTP would be unsuitable. Current implementation is HTTP and that makes sense, just thinking about the options in the future.